lib/analytics.js

```diff
@@ -0,0 +1,115 @@
+const fs = require('fs');
+
+// We can either use a process-wide object regularly saved to a JSON file,
+// or a Redis equivalent (for multi-process / when the filesystem is unreliable.
+var redis;
+var useRedis = false;
```

Comment:
I moved all the code as is. If it's alright with you, I'd rather move it first, and then refactor / rewrite it, addressing these comments. There's a lot here that can be improved.